odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1953,56 +1925,12 @@ def is_node_branded(self, node):
             for attr in node.attrib
         )
 
-    @tools.ormcache('self.id')
-    def get_view_xmlid(self):
-        domain = [('model', '=', 'ir.ui.view'), ('res_id', '=', self.id)]
-        xmlid = self.env['ir.model.data'].sudo().search_read(domain, ['module', 'name'])[0]
-        return '%s.%s' % (xmlid['module'], xmlid['name'])
-
     @api.model
     def render_public_asset(self, template, values=None):
```

Comment:
Security alert: this does the same thing as the method below but without the `_`

Reply:
@mart-e @odony  This seems strange to me, it is necessary to keep only render_template in public?
`render_public_asset` vs `_render_template`

Reply:
@Gorash how about the `check_view_access` ?

Reply:
@mart-e it is a call in a controller, the `check_view_access` can be done before calling `_render_template`.